src/plots/piePlot.ts

```diff
@@ -209,6 +233,50 @@ export module Plots {
       var avgAngle = (startAngle + endAngle) / 2;
       return { x: avgRadius * Math.sin(avgAngle), y: -avgRadius * Math.cos(avgAngle) };
     }
+
+    protected _additionalPaint(time: number) {
+      if (this._labelsEnabled) {
+        Utils.Window.setTimeout(() => this._drawLabels(), time);
+      }
+    }
+
+    private _drawLabels() {
+      var attrToProjector = this._generateAttrToProjector();
+      var labelArea = this._renderArea.append("g").classed("label-area", true);
+      var measurer = new SVGTypewriter.Measurers.CacheCharacterMeasurer(labelArea);
+      var writer = new SVGTypewriter.Writers.Writer(measurer);
+      this.entities().forEach((entity) => {
```

Comment:
Why use `entities()` inside here? It seems like the `Plot` should already be able to get the necessary values by other means right?